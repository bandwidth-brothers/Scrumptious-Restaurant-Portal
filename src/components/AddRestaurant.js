import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import Alert from '@material-ui/lab/Alert';
import React, { useState, useContext } from 'react';
import { useHistory } from 'react-router-dom';
import AuthService from '../services/AuthService';
import { RestaurantService } from '../services/RestaurantService';
import { RestaurantsStateContext } from '../pages/Admin';

const useStyles = makeStyles((theme) => ({
    layout: {
        width: 'auto',
        marginLeft: theme.spacing(2),
        marginRight: theme.spacing(2),
        [theme.breakpoints.up(600 + theme.spacing(2) * 2)]: {
            width: 600,
            marginLeft: 'auto',
            marginRight: 'auto',
        },
    },
    paper: {
        marginTop: theme.spacing(3),
        marginBottom: theme.spacing(3),
        padding: theme.spacing(2),
        [theme.breakpoints.up(600 + theme.spacing(3) * 2)]: {
            width: 450,
            marginTop: theme.spacing(6),
            marginBottom: theme.spacing(6),
            padding: theme.spacing(3),
        },
    },
    buttons: {
        display: 'flex',
        justifyContent: 'flex-end',
    },
    button: {
        marginTop: theme.spacing(3),
        marginLeft: theme.spacing(1),
    },
}));

function AddRestaurant() {
    const classes = useStyles();
    const history = useHistory();
    const auth = AuthService.getCurrentUser();

    const initialRestauState = {
        name: "",
        lineOne: "",
        lineTwo: "",
        city: "",
        state: "",
        zip: "",
        restaurantOwnerId: ""
    };

    const [restau, setRestau] = useState(initialRestauState);
    const [alertContent, setAlertContent] = useState('');
    const [alert, setAlert] = useState(false);
    const [restaurants, setRestaurants] = useContext(RestaurantsStateContext);

    const handleInputChange = e => {
        const { name, value } = e.target;
        setRestau({ ...restau, [name]: value });
    };

    const isValid = (restau) => {
        if (restau.name === "") {
            setAlertContent("restaurant name is empty");
            setAlert(true);
            return false;
        }

        if (restau.lineOne === "") {
            setAlertContent("address line one is empty");
            setAlert(true);
            return false;
        }

        if (restau.city === "") {
            setAlertContent("address city is empty");
            setAlert(true);
            return false;
        }

        if (restau.state === "") {
            setAlertContent("address state is empty");
            setAlert(true);
            return false;
        }

        if (restau.zip === "" || isNaN(restau.zip)) {
            setAlertContent("address zip is not valid");
            setAlert(true);
            return false;
        }

        return true;
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (!isValid(restau)) {
            return;
        }

        


        if (auth) {
            setRestau({ ...restau, "restaurantOwnerId": auth.userId });
            RestaurantService.createRestaurant(restau)
                .then(function (response) {

                    RestaurantService.getRestaurantList(auth.userId)
                    .then(function (r) {
                        const data = r.data;
                        setRestaurants(data);
                        history.push("/admin/list");
                    })                      

                })
                .catch(function (error) {
                    console.log(error);
                });

        }
    };

    return (
        <React.Fragment>
            <CssBaseline />
            <main className={classes.layout}>
                <Paper className={classes.paper}>
                    <Typography component="h1" variant="h5" align="center">
                        Add Restaurant
                    </Typography>

                    <React.Fragment>
                        {(
                            <form >
                                <React.Fragment>
                                    <Grid container spacing={3}>
                                        <Grid item xs={12}>
                                            <TextField
                                                required
                                                id="storeName"
                                                name="name"
                                                label="Restaurant Name"
                                                fullWidth
                                                onChange={handleInputChange}
                                                autoComplete="store-name"
                                            />
                                        </Grid>
                                        <Grid item xs={12}>
                                            <TextField
                                                required
                                                id="address1"
                                                name="lineOne"
                                                label="Address line 1"
                                                fullWidth
                                                value={restau.lineOne}
                                                onChange={handleInputChange}
                                            />
                                        </Grid>
                                        <Grid item xs={12}>
                                            <TextField
                                                id="address2"
                                                name="lineTwo"
                                                label="Address line 2"
                                                fullWidth
                                                onChange={handleInputChange}
                                            />
                                        </Grid>
                                        <Grid item xs={12} sm={6}>
                                            <TextField
                                                required
                                                id="city"
                                                name="city"
                                                label="City"
                                                fullWidth
                                                onChange={handleInputChange}
                                            />
                                        </Grid>
                                        <Grid item xs={12} sm={6}>
                                            <TextField
                                                required
                                                id="state"
                                                name="state"
                                                label="State"
                                                fullWidth
                                                onChange={handleInputChange}
                                            />
                                        </Grid>
                                        <Grid item xs={12} sm={6}>
                                            <TextField
                                                required
                                                id="zip"
                                                name="zip"
                                                label="Zip code"
                                                fullWidth
                                                onChange={handleInputChange}
                                            />
                                        </Grid>
                                        <Grid item xs={12} sm={6}>
                                            <TextField
                                                id="country"
                                                name="country"
                                                label="US"
                                                // defaultValue="US"
                                                disabled
                                                fullWidth
                                            />
                                        </Grid>
                                        <Grid item xs={12}>
                                            {alert ? <Alert severity='error'>{alertContent}</Alert> : <></>}
                                        </Grid>
                                    </Grid>
                                </React.Fragment>

                                <div className={classes.buttons}>
                                    <Button
                                        variant="contained"
                                        color="primary"
                                        onClick={handleSubmit}
                                        className={classes.button}
                                    >
                                        ADD
                                    </Button>
                                </div>
                            </form>
                        )}
                    </React.Fragment>
                </Paper>
            </main>
        </React.Fragment>
    );
}
export default AddRestaurant;