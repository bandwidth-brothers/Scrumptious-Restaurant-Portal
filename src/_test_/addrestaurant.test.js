import { cleanup, render, screen } from '@testing-library/react'
import { shallow } from "enzyme"
import React from 'react'
import AddRestaurant from '../components/AddRestaurant'

describe('<AddRestaurant /> with no props', () => {

    jest.mock('../services/AuthService');
    const container = shallow(<AddRestaurant />);
    afterEach(cleanup)

    it("should render initial layout", () => {

        expect(container.getElements()).toMatchSnapshot();
    });

    it('render form elements', () => {
        render(<AddRestaurant />);
        expect(screen.getByText('Restaurant Name')).toBeInTheDocument();
        expect(screen.getByText('Address line 1')).toBeInTheDocument();
        expect(screen.getByText('Address line 2')).toBeInTheDocument();
        expect(screen.getByText('City')).toBeInTheDocument();
        expect(screen.getByText('State')).toBeInTheDocument();
        expect(screen.getByText('Zip code')).toBeInTheDocument();
        expect(screen.getByText('US')).toBeInTheDocument();
        expect(screen.getByText('ADD')).toBeInTheDocument();
    });

})