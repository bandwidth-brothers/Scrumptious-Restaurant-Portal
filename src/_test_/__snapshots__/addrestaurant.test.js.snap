// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddRestaurant /> with no props should render initial layout 1`] = `
Array [
  <React.Fragment>
    <WithStyles(CssBaseline) />
    <main
      className="makeStyles-layout-1"
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-2"
      >
        <WithStyles(ForwardRef(Typography))
          align="center"
          component="h1"
          variant="h4"
        >
          Add Restaurant
        </WithStyles(ForwardRef(Typography))>
        <React.Fragment>
          <form>
            <React.Fragment>
              <WithStyles(ForwardRef(Grid))
                container={true}
                spacing={3}
              >
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  xs={12}
                >
                  <WithStyles(ForwardRef(TextField))
                    autoComplete="store-name"
                    fullWidth={true}
                    id="storeName"
                    label="Restaurant Name"
                    name="name"
                    onChange={[Function]}
                    required={true}
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  xs={12}
                >
                  <WithStyles(ForwardRef(TextField))
                    fullWidth={true}
                    id="address1"
                    label="Address line 1"
                    name="lineOne"
                    onChange={[Function]}
                    required={true}
                    value=""
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  xs={12}
                >
                  <WithStyles(ForwardRef(TextField))
                    fullWidth={true}
                    id="address2"
                    label="Address line 2"
                    name="lineTwo"
                    onChange={[Function]}
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  sm={6}
                  xs={12}
                >
                  <WithStyles(ForwardRef(TextField))
                    fullWidth={true}
                    id="city"
                    label="City"
                    name="city"
                    onChange={[Function]}
                    required={true}
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  sm={6}
                  xs={12}
                >
                  <WithStyles(ForwardRef(TextField))
                    fullWidth={true}
                    id="state"
                    label="State"
                    name="state"
                    onChange={[Function]}
                    required={true}
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  sm={6}
                  xs={12}
                >
                  <WithStyles(ForwardRef(TextField))
                    fullWidth={true}
                    id="zip"
                    label="Zip code"
                    name="zip"
                    onChange={[Function]}
                    required={true}
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  sm={6}
                  xs={12}
                >
                  <WithStyles(ForwardRef(TextField))
                    disabled={true}
                    fullWidth={true}
                    id="country"
                    label="US"
                    name="country"
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  xs={12}
                >
                  <React.Fragment />
                </WithStyles(ForwardRef(Grid))>
              </WithStyles(ForwardRef(Grid))>
            </React.Fragment>
            <div
              className="makeStyles-buttons-3"
            >
              <WithStyles(ForwardRef(Button))
                className="makeStyles-button-4"
                color="primary"
                onClick={[Function]}
                variant="contained"
              >
                ADD
              </WithStyles(ForwardRef(Button))>
            </div>
          </form>
        </React.Fragment>
      </WithStyles(ForwardRef(Paper))>
    </main>
  </React.Fragment>,
]
`;
